### PR TITLE
Rename React methods

### PR DESCRIPTION
**Pull request by GeorgiSTodorov**
Everything should still work the same way as before.

EDIT: please review commit-by-commit as there is a noise change in between that has been introduced to shut up the `StoreMixin` (and make it fully functional again).

You'll want to merge https://github.com/mesosphere/dcos-ui-plugins-private/pull/1057 along.

EDIT 2:
sorry for delaying this but it turned out that seemingly we can get rid of ALL `super.UNSAFE_*`s and the `SaveStateMixin`! 🎉

### CODE REVIEW

File: plugins/nodes/src/js/pages/nodes/NodeDetailPage.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ class NodeDetailPage extends mixin(TabsMixin, StoreMixin) {
     this.updateCurrentTab(nextProps);
   }
 
-  componentWillUpdate() {
+  UNSAFE_componentWillUpdate() {
     super.componentWillUpdate(...arguments);
```
Comment (pierrebeitz) [inline plugins/nodes/src/js/pages/nodes/NodeDetailPage.js]: shouldn't we change those calls as well?

Comment (GeorgiSTodorov) [reply plugins/nodes/src/js/pages/nodes/NodeDetailPage.js]: We should, this PR is still work in progress.

**Comment by GeorgiSTodorov** [comment]
I am looking into the failing unit tests.

**Comment by GeorgiSTodorov** [comment]
Rebased.

**Comment by GeorgiSTodorov** [comment]
For some reason, having `super.UNSAFE_` breaks some pages, for example Networking > Networks and Package Detail. @pierrebeitz , do you know why this might break? Also, it leads to some weird unit test failures.

**Comment by GeorgiSTodorov** [comment]
Thanks, @pierrebeitz .

**Pull request merged**